cmake/RepositoryDependenciesSetup.cmake

```diff
@@ -10,4 +10,77 @@ if (TPL_ENABLE_CUDA AND NOT Kokkos_ENABLE_Cuda_Relocatable_Device_Code)
   elseif (${PROJECT_NAME}_ENABLE_ShyLU_NodeTacho)
     message(FATAL_ERROR "ERROR: ${PROJECT_NAME}_ENABLE_ShyLU_NodeTacho=ON but TPL_ENABLE_CUDA='${TPL_ENABLE_CUDA}' AND Kokkos_ENABLE_Cuda_Relocatable_Device_Code='${Kokkos_ENABLE_Cuda_Relocatable_Device_Code}' which is not allowed!")
   endif()
-endif() 
\ No newline at end of file
+endif() 
+
+#########################################################################
+# STKBalance does not work with GO=INT or GO=UNSIGNED
+# Warn and disable it if it was set as a dependence of STK
+# Error out if it was explicitly requested by user
+
+IF ("${Tpetra_ENABLE_DEPRECATED_CODE}" STREQUAL "OFF")  # Remove this test when
+                                                        # Tpetra DEPRECATED
+                                                        # code is removed
+  SET(KDD_ENABLE_DEPRECATED OFF)  # Remove this line when DEPRECATED is removed
+  SET(KDD_INT_INT OFF)     # Current default
+ELSE() # Remove this "else" section when Tpetra DEPRECATED code is removed
+  SET(KDD_ENABLE_DEPRECATED ON)   # Remove this line when DEPRECATED is removed
+  SET(KDD_INT_INT ON)      # Deprecated default
+ENDIF()
+
+SET(KDD_INT_UNSIGNED OFF)  # Current default
+SET(KDD_INT_LONG OFF)      # Current default
+SET(KDD_INT_LONG_LONG ON)  # Current default
+
+IF (NOT ("${Tpetra_INST_INT_INT}" STREQUAL ""))
+  SET(KDD_INT_INT ${Tpetra_INST_INT_INT})
+  if (${KDD_INT_INT}                      # Keep this test but
+      AND (NOT ${KDD_ENABLE_DEPRECATED})) # remove this test when DEPRECATED
+                                          # is removed
+    SET(KDD_INT_LONG_LONG OFF)
+  ENDIF()
+ENDIF()
+
+IF(NOT ("${Tpetra_INST_INT_UNSIGNED}" STREQUAL ""))
+  SET(KDD_INT_UNSIGNED ${Tpetra_INST_INT_UNSIGNED})
+  if (${KDD_INT_UNSIGNED}                 # Keep this test but
+      AND (NOT ${KDD_ENABLE_DEPRECATED})) # remove this test when DEPRECATED
+                                          # is removed
+    SET(KDD_INT_LONG_LONG OFF)
+  ENDIF()
+ENDIF()
+
+IF(NOT ("${Tpetra_INST_INT_LONG}" STREQUAL ""))
+  SET(KDD_INT_LONG ${Tpetra_INST_INT_LONG})
+  if (${KDD_INT_LONG}                     # Keep this test but
+      AND (NOT ${KDD_ENABLE_DEPRECATED})) # remove this test when DEPRECATED
+                                          # is removed
+    SET(KDD_INT_LONG_LONG OFF)
+  ENDIF()
+ENDIF()
+
+IF(NOT ("${Tpetra_INST_INT_LONG_LONG}" STREQUAL ""))
+  SET(KDD_INT_LONG_LONG ${Tpetra_INST_INT_LONG_LONG})
+ENDIF()
+
+IF ((NOT ${KDD_INT_LONG}) AND (NOT ${KDD_INT_LONG_LONG}))
+  IF ("${${PROJECT_NAME}_ENABLE_STKBalance}" STREQUAL "")
+    # STKBalance may be enabled but only implicitly (as a dependence of STK);
+    # give a warning but turn off STKBalance support
+    MESSAGE(WARNING "int global indices are enabled in Trilinos. "
```

Comment:
Should this really be a `MESSAGE(WARNING ...)` instead of a `MESSAGE("NOTE: " ...)`?  I know that some projects like Charon2 really don't like CMake warnings unless they are real warnings.  But I guess they can make this go away by just explicitly setting `-DTrilinos_ENABLE_STKBalance=OFF`, right?

Reply:
Just to be clear, if I set `-DTpetra_INST_INT_INT=OFF`, then I should not get this warning?

Reply:
Right, with either Trilinos_ENABLE_STKBalance=OFF or Tpetra_INST_INT_INT=OFF, there would be no message.
Also, there's no message in the current default Trilinos configuration (with both GO=int and GO=long long) or the soon-to-be default configuration (with GO=long long) since long long GOs are available.  
That said, if you say "NOTE" is better than "WARNING," I am fine with that.  I'll follow your judgement on CMake etiquette.